### PR TITLE
Added the feature to resurrect extinct species found in ancient ruins…

### PR DESCRIPTION
**Pull request by Deepsloth**
… with the Xenoresurrection Lab.

### CODE REVIEW

**Comment by Deepsloth** [comment]
Information: http://www.freeorion.org/forum/viewtopic.php?f=15&t=6570&start=26

AI doesn't need any immediate updates, but should use the building in the future (and then the Banforo will need a little adjustment for their light sensivity).

File: default/stringtables/en.txt [hunk 1]
```diff
@@ -5504,6 +5504,64 @@ Reason for Staying:
...
+Prefer Barren planets.
+[[encyclopedia LITHIC_SPECIES_TITLE]]
+'''
+
+SP_BANFORO_DESC
+'''Description:
+A Banfo looks like a teddy bear made of white stone with two large black crystal eyes. It's likely that the Banforo species was artificially created, but information that could show proof was lost in history. The eyes are their most important organs, not only do they serve for perception, but also for communication by showing light patterns around their frames. Banforo can distinguish colors and patterns of even very dim light, as well as focus on light sources far away. Very bright light on the other hand is very hard on their eyes and disturbs their communication. Thus the evening is the favorite time for the Banforo and a defining ritual of their civilization is collectively gazing at the stars once the sun has set.
+
+Social Structure:
+The Banforo are very social towards each other and share strong bonds. Ressources are always shared and private ownership doesn't exist for goods with material value. The form of government can be called communistic.
```
Comment (MatGB) [inline default/stringtables/en.txt]: Resources

File: default/stringtables/en.txt [hunk 2]
```diff
@@ -9063,6 +9145,12 @@ Starlane Nexus
 BLD_STARLANE_NEXUS_DESC
 Generates new starlanes to other star systems that can be connected without crossing existing starlanes or passing too close to another system.
 
+BLD_XENORESURRECTION_LAB
+Xenoresurrection Lab
+
+BLD_XENORESURRECTION_LAB_DESC
+The Xenoresurrection Lab restores living samples of extinct species, allowing colonization of supply connected planets with the extinct species found in [[special ANCIENT_RUINS_SPECIAL]]. It can be build on planets where the excavations of [[special ANCIENT_RUINS_SPECIAL]] have found well preserved bodies of an exstinct species.
```
Comment (MatGB) [inline default/stringtables/en.txt]: extinct

File: default/specials.txt [hunk 3]
```diff
@@ -290,7 +290,7 @@ Special
             scope = Source
             activation = OwnerHasTech name = "LRN_XENOARCH"
             effects = [
-                AddSpecial name = "ANCIENT_RUINS_DEPLETED_SPECIAL"
+                AddSpecial name = OneOf("ANCIENT_RUINS_DEPLETED_SPECIAL", "ANCIENT_RUINS_DEPLETED_BANFORO_SPECIAL", "ANCIENT_RUINS_DEPLETED_KILANDOW_SPECIAL", "ANCIENT_RUINS_DEPLETED_MISIORLA_SPECIAL")
```
Comment (MatGB) [inline default/specials.txt]: OK, so the plan is to have a 1 in 4 chance of nothing and a 3 in 4 chance of one of the new guys regardless of other results?

**Comment by MatGB** [comment]
I love this idea, glad you've had time for it, and you've put way more work in than I would've, I'd have just had them have crashed cryonic colony pods and create a ship in the same way as the Dragon Tooth.

I wasn't sure about the use of "good chance" in the ruins description but if I'm reading the script right it actually is an accurate description and although I don't like the language I can't think of anything better. 

Fix the typos I'll merge it, I need to recompile at some point soon anyway and these guys look fun.

**Comment by Dilvish-fo** [comment]
Rather than add a new version of ANCIENT_RUINS_DEPLETED_X for every species, it seems to me it would be a much better structure to leave the ANCIENT_RUINS_DEPLETED as it was, and simply add a new special for the possibility of resurrecting the given species.  That seems like a much cleaner (and more flexible) way to maintain things, both from the pure scripting perspective, and from the AI adjustment perspective.  

As it stands this would break the AI's understanding of ANCIENT_RUINS_DEPLETED, for no good reason it seems to me.  The adjustment for that limited aspect of it wouldn't be hard, but it just doesn't seem like a good structure to me.

**Comment by Deepsloth** [comment]
That's a good idea Dilvish. That would indeed be a lot cleaner and flexible.
I will create an update when i find some time.

**Comment by Deepsloth** [comment]
Update has been added. In addition to the change suggested by Dilvish, i've also added SitReps (including Rumors) for finding extinct species and updated the Kilandow icon.

**Comment by wheals** [comment]
Looks like the special icons should be in `icons/specials_large`, rather than `huge`, since they're not as big as the rest in that folder.

**Comment by Deepsloth** [comment]
Since the icons are downsampled by the engine the specials icon folders other than huge are obsolete. They should probably be removed at one point in the future (and the specials_huge folder can be renamed to just specials).
I don't think it's a good idea to put new icons in obsolete folders.

**Comment by MatGB** [comment]
@Dilvish-fo , just checking you're OK with this going in, I'm going through old PRs and this one looks good to me, testing it a bit at the moment.

**Comment by Dilvish-fo** [comment]
I haven't tested it, but have no objection to it now that it's been restructured (and I agree the idea is nice).

**Comment by MatGB** [comment]
Cool, I'm playing now and got Misiorla first off, given I had 4 Mu Ursh homeworlds that wasn't as big a boost, it looks fine, I want to do a few tweaks, give the sitreps a priority, then I'll commit the lot.  I don't have a strong opinion on the balance of the species yet but that can come later, species overall need a proper balance pass once the priority system is bedded in (which I'll get back to soon).

File: default/specials.txt [hunk 4]
```diff
@@ -1931,5 +2011,23 @@ CHANCE_OF_GUARD_1
...
+
+        EffectsGroup
+            scope = And [
+                Source
+                Focus type = "FOCUS_RESEARCH"
+            ]
+            effects = SetTargetResearch value = Value + Target.Population * 5 * [[RESEARCH_PER_POP]]
+        ]
+    graphic = "icons/specials_huge/ancient_ruins_excavated.png"
+'''
```
Comment (Vezzra) [inline default/specials.txt:2030]: I'm probably missing something, but after going over all the changes of this PR I can't find where this macro is being used... so what's the purpose of this new macro? Or is that a leftover of the first draft, where you had several versions of the ANCIENT_RUINS_DEPLETED_SPECIAL?

**Comment by Vezzra** [comment]
Looks good as far as I can see :smiley:

File: default/specials.txt (hunk 4 above)
Comment (MatGB) [inline default/specials.txt:2030]: Good catch, I'm finishing off a few polishes to it to commit at the same time (minor tweaks in wording, sitrep priorities), it does appear to not being used, I'll confirm that and delete it

**Comment by MatGB** [comment]
Closed with [3e239a1](https://github.com/freeorion/freeorion/commit/3e239a1763e6783b35268ff8bf2bc22404e05c37)

File: default/specials.txt (hunk 4 above)
Comment (MatGB) [inline default/specials.txt:2030]: Gah, got distracted, committed then saw this, I'll fix tomorrow.

**Pull request closed without merging**